src/parser/mod.rs

```diff
@@ -7220,8 +7220,35 @@ impl<'a> Parser<'a> {
         // Clickhouse has `ON CLUSTER 'cluster'` syntax for DDLs
         let on_cluster = self.parse_optional_on_cluster()?;
 
-        let like = if self.parse_keyword(Keyword::LIKE) || self.parse_keyword(Keyword::ILIKE) {
-            self.parse_object_name(allow_unquoted_hyphen).ok()
+        // Try to parse `CREATE TABLE new (LIKE old [{INCLUDING | EXCLUDING} DEFAULTS])` or `CREATE TABLE new LIKE old`
+        let like = if self.dialect.supports_create_table_like_in_parens()
```

Comment:
Can we pull the logic out to a function like `maybe_parse_create_table_like()`? since it now adds a bit of code to the existing function